### PR TITLE
Allow enabling modular headers for pods in the Podfile

### PR DESCRIPTION
**Pull request by segiddins**

### CODE REVIEW

File: lib/cocoapods-core/podfile/target_definition.rb [hunk 1]
```diff
@@ -457,6 +457,96 @@ def all_whitelisted_configurations
...
+            # Clean pods that are set to not inhibit inside parent if inhibit_all_warnings! was set.
+            parent_hash['not_for_pods'] = nil
+          end
+          parent_hash.merge(raw_hash) do |_, l, r|
+            Array(l).concat(r).uniq
+          end
+        end
+      end
+
+      # @return [Bool] whether the target definition should inhibit warnings
```
Comment (dnkoutso) [inline lib/cocoapods-core/podfile/target_definition.rb]: copy pasta `inhibit warnings` in the documentation

Comment (dnkoutso) [reply lib/cocoapods-core/podfile/target_definition.rb]: same below

File: lib/cocoapods-core/podfile/target_definition.rb [hunk 2]
```diff
@@ -457,6 +457,96 @@ def all_whitelisted_configurations
...
+        elsif use_modular_headers_hash['all']
+          true
+        elsif !root? && parent.build_pod_as_module?(pod_name)
+          true
+        else
+          Array(use_modular_headers_hash['for_pods']).include? pod_name
+        end
+      end
+
+      # Sets whether the target definition should inhibit the warnings during
```
Comment (dnkoutso) [inline lib/cocoapods-core/podfile/target_definition.rb]: same

File: lib/cocoapods-core/podfile/target_definition.rb [hunk 3]
```diff
@@ -457,6 +457,96 @@ def all_whitelisted_configurations
...
+      # @param  [Bool] flag
+      #         Whether the warnings should be suppressed.
+      #
+      # @return [void]
+      #
+      def use_modular_headers_for_all_pods=(flag)
+        raw_use_modular_headers_hash['all'] = flag
+      end
+
+      # Inhibits warnings for a specific pod during compilation.
```
Comment (dnkoutso) [inline lib/cocoapods-core/podfile/target_definition.rb]: same

File: lib/cocoapods-core/podfile/target_definition.rb [hunk 4]
```diff
@@ -873,6 +965,28 @@ def parse_inhibit_warnings(name, requirements)
         requirements.pop if options.empty?
       end
 
+      # Removes :inhibit_warnings from the requirements list, and adds
```
Comment (dnkoutso) [inline lib/cocoapods-core/podfile/target_definition.rb]: same

File: lib/cocoapods-core/podfile/target_definition.rb [hunk 5]
```diff
@@ -457,6 +457,96 @@ def all_whitelisted_configurations
...
+            Array(l).concat(r).uniq
+          end
+        end
+      end
+
+      # @return [Bool] whether the target definition should inhibit warnings
+      #         for a single pod. If inhibit_all_warnings is true, it will
+      #         return true for any asked pod.
+      #
+      def build_pod_as_module?(pod_name)
```
Comment (dnkoutso) [inline lib/cocoapods-core/podfile/target_definition.rb]: is this more like a `build_module_for_pod?(pod_name)`?

File: spec/podfile/dsl_spec.rb [hunk 6]
```diff
@@ -207,6 +207,155 @@ module Pod
...
+                pod 'PonyDebugger', :modular_headers => false
+              end
+            end
+          end
+
+          target = podfile.target_definitions['App']
+          target.build_pod_as_module?('PonyDebugger').should.be.true
+        end
+
+        it 'allows inhibiting in parent and not inhibiting in child on a single dependency' do
```
Comment (dnkoutso) [inline spec/podfile/dsl_spec.rb]: test name more about inhibiting than fixing modular headers

**Comment by dnkoutso** [comment]
functionally looks good to me. Just needs to update all doc and tests so they are not a copy paste from `inhibit_warnings` DSL

**Comment by dnkoutso** [comment]
Approved. Can merge after fixing the docs etc.

File: lib/cocoapods-core/podfile/target_definition.rb [hunk 7]
```diff
@@ -457,6 +457,96 @@ def all_whitelisted_configurations
...
       #--------------------------------------#
 
+      def raw_use_modular_headers_hash
+        get_hash_value('use_modular_headers', {})
+      end
+      private :raw_use_modular_headers_hash
+
+      # Returns the use_modular_headers hash pre-populated with default values.
+      #
+      # @return [Hash<String, Array>] Hash with :all key for inhibiting all
```
Comment (paulb777) [inline lib/cocoapods-core/podfile/target_definition.rb]: warnings copy pasta issue in this header comment

File: lib/cocoapods-core/podfile/target_definition.rb [hunk 8]
```diff
@@ -457,6 +457,96 @@ def all_whitelisted_configurations
...
+      # @return [void]
+      #
+      def use_modular_headers_for_all_pods=(flag)
+        raw_use_modular_headers_hash['all'] = flag
+      end
+
+      # Inhibits warnings for a specific pod during compilation.
+      #
+      # @param  [String] pod_name
+      #         Name of the pod for which the warnings will be inhibited or not.
```
Comment (paulb777) [inline lib/cocoapods-core/podfile/target_definition.rb]: same

File: lib/cocoapods-core/podfile/target_definition.rb (hunk 5 above)
Comment (segiddins) [reply lib/cocoapods-core/podfile/target_definition.rb]: not really

File: lib/cocoapods-core/podfile/target_definition.rb [hunk 9]
```diff
@@ -457,6 +457,96 @@ def all_whitelisted_configurations
...
+      #
+      # @param  [Bool] should_inhibit
+      #         Whether the warnings should be inhibited or not for given pod.
```
Comment (paulb777) [inline lib/cocoapods-core/podfile/target_definition.rb]: same

File: lib/cocoapods-core/podfile/target_definition.rb [hunk 10]
```diff
@@ -873,6 +965,28 @@ def parse_inhibit_warnings(name, requirements)
         requirements.pop if options.empty?
       end
 
+      # Removes :inhibit_warnings from the requirements list, and adds
+      # the pod's name into internal hash for disabling warnings.
+      #
+      # @param [String] pod name
+      #
+      # @param [Array] requirements
+      #        If :inhibit_warnings is the only key in the hash, the hash
```
Comment (paulb777) [inline lib/cocoapods-core/podfile/target_definition.rb]: same

File: lib/cocoapods-core/podfile/target_definition.rb [hunk 11]
```diff
@@ -873,6 +965,28 @@ def parse_inhibit_warnings(name, requirements)
         requirements.pop if options.empty?
       end
 
+      # Removes :inhibit_warnings from the requirements list, and adds
+      # the pod's name into internal hash for disabling warnings.
```
Comment (paulb777) [inline lib/cocoapods-core/podfile/target_definition.rb]: same

File: spec/podfile/target_definition_spec.rb [hunk 12]
```diff
@@ -305,6 +305,61 @@ module Pod
...
+        @parent.should.build_pod_as_module?('RestKit')
+      end
+
+      it 'does not use modular headers per pod if the option is false' do
+        @parent.inhibit_all_warnings = true
+        @parent.store_pod('ASIHTTPRequest', :modular_headers => false)
+        @parent.should.not.build_pod_as_module?('ASIHTTPRequest')
+      end
+
+      it 'must delete the hash if it was empty. otherwise breaks Dependency' do
```
Comment (paulb777) [inline spec/podfile/target_definition_spec.rb]: period. lower case

File: lib/cocoapods-core/podfile/target_definition.rb (hunk 10 above)
Comment (dnkoutso) [reply lib/cocoapods-core/podfile/target_definition.rb]: yeap just this one left

**Comment by segiddins** [comment]
Feedback addressed, sorry for the bad copy-pasta

**Pull request merged**